pkg/auth/handler/session/revoke.go

```diff
@@ -89,98 +73,51 @@ const RevokeRequestSchema = `
 		@Response 200 {EmptyResponse}
 */
 type RevokeHandler struct {
-	Validator        *validation.Validator      `dependency:"Validator"`
-	RequireAuthz     handler.RequireAuthz       `dependency:"RequireAuthz"`
-	TxContext        db.TxContext               `dependency:"TxContext"`
-	SessionProvider  session.Provider           `dependency:"SessionProvider"`
-	IdentityProvider principal.IdentityProvider `dependency:"IdentityProvider"`
-	AuthInfoStore    authinfo.Store             `dependency:"AuthInfoStore"`
-	UserProfileStore userprofile.Store          `dependency:"UserProfileStore"`
-	HookProvider     hook.Provider              `dependency:"HookProvider"`
+	validator      *validation.Validator
+	txContext      db.TxContext
+	sessionManager sessionRevokeManager
 }
 
-func (h RevokeHandler) ProvideAuthzPolicy() authz.Policy {
+func (h *RevokeHandler) ProvideAuthzPolicy() authz.Policy {
 	return policy.RequireValidUser
 }
 
-func (h RevokeHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	var response handler.APIResponse
+func (h *RevokeHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	session := auth.GetSession(r.Context())
+
 	var payload RevokeRequestPayload
-	// TODO(authn): use correct session ID
-	payload.CurrentSessionID = ""
-	if err := handler.BindJSONBody(r, w, h.Validator, "#SessionRevokeRequest", &payload); err != nil {
-		response.Error = err
-	} else {
-		result, err := h.Handle(r, payload)
-		if err != nil {
-			response.Error = err
-		} else {
-			response.Result = result
-		}
+	payload.CurrentSessionID = session.SessionID()
```

Comment:
Is CurrentSessionID being respected?

Reply:
It is validated in `Validate()` function of the payload.